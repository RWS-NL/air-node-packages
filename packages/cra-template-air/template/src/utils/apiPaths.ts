export enum API_METHOD {
  GET = 'get',
  POST = 'post',
  PUT = 'put',
  DELETE = 'delete'
}

export enum ENDPOINTS {
  CONFIG = 'config'
}

export enum API_VERSION {
  v1 = 'v1'
}

export enum ACTION_ENDPOINTS {
  GET_CONFIG = 'getConfig'
}

export const CONFIG_PATH = (version: API_VERSION) => `/api/${version}/${ENDPOINTS.CONFIG}`;
