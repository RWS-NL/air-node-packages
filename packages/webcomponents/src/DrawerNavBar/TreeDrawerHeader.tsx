import Divider from '@material-ui/core/Divider';
import FormControl from '@material-ui/core/FormControl';
import Grid from '@material-ui/core/Grid';
import IconButton from '@material-ui/core/IconButton';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';
import MenuItem from '@material-ui/core/MenuItem';
import Select from '@material-ui/core/Select';
import { createStyles, makeStyles, Theme } from '@material-ui/core/styles';
import ArrowBackIcon from '@material-ui/icons/ArrowBack';
import ArrowForward from '@material-ui/icons/ArrowForward';
import InfoIcon from '@material-ui/icons/Info';
import clsx from 'clsx';
import React, { memo } from 'react';
import { Else, If, Then, When } from 'react-if';
import { Tooltip } from '../Tooltip';
import { TreeDrawerProps } from './DrawerProps';

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    treeDropdownGrid: {
      maxHeight: '1rem',
      marginTop: '-2.5rem'
    },
    treeSelector: {
      margin: theme.spacing(1),
      minWidth: 120
    },
    treeDrawerCloseIcon: {
      marginRight: theme.spacing(1)
    },
    smallInfoButton: {
      width: '0.6em',
      height: '0.6em'
    },
    treeInfoButton: {
      paddingTop: 0,
      paddingBottom: 6
    },
    closeListItem: {
      justifyContent: 'flex-end'
    },
    divider: {
      backgroundColor: '#CCE1EB' // Dark Blue 20
    }
  })
);

export const TreeDrawerHeader = memo((treeDrawer: TreeDrawerProps) => {
  const classes = useStyles();

  return (
    <>
      <List>
        <ListItem
          button
          disableGutters={treeDrawer.open}
          onClick={treeDrawer.toggleDrawer}
          classes={{
            root: clsx({ [classes.closeListItem]: treeDrawer.open })
          }}
        >
          <Grid
            container
            direction='row'
            justify='space-between'
            alignItems='center'
            alignContent='stretch'
            wrap='nowrap'
            spacing={0}
          >
            <When condition={treeDrawer.open}>
              <Grid item xs={1} />
              <Grid item xs={8} classes={{ root: classes.treeDropdownGrid }}>
                <ListItemText
                  disableTypography
                  primary={
                    <FormControl className={classes.treeSelector} size='small'>
                      <Select
                        value={treeDrawer.currentDropdownValue}
                        onChange={(event) => {
                          event.preventDefault();
                          treeDrawer.hanbdleDropdownChange(event);
                          event.stopPropagation();
                        }}
                        onClose={(event) => {
                          event.preventDefault();
                          event.stopPropagation();
                        }}
                        onOpen={(event) => {
                          event.preventDefault();
                          event.stopPropagation();
                        }}
                        fullWidth
                        displayEmpty
                        disableUnderline
                        MenuProps={{
                          getContentAnchorEl: null,
                          anchorOrigin: {
                            vertical: 'bottom',
                            horizontal: 'center'
                          },
                          transformOrigin: {
                            vertical: 'top',
                            horizontal: 'center'
                          },
                          elevation: 2
                        }}
                        renderValue={(value) => <b>{value as string}</b>}
                      >
                        {treeDrawer.dropdownValues.map(({ value, label }, index) => (
                          <MenuItem key={index} value={value}>
                            {label}
                          </MenuItem>
                        ))}
                      </Select>
                    </FormControl>
                  }
                />
              </Grid>
              <Grid item xs={1}>
                <Tooltip title={treeDrawer.tooltipText}>
                  <IconButton color='primary' size='small' classes={{ root: classes.treeInfoButton }}>
                    <InfoIcon className={classes.smallInfoButton} />
                  </IconButton>
                </Tooltip>
              </Grid>
            </When>
            <Grid item xs={1} classes={{ root: classes.treeDrawerCloseIcon }}>
              <ListItemIcon>
                <If condition={treeDrawer.open}>
                  <Then>
                    <ArrowBackIcon color='primary' />
                  </Then>
                  <Else>
                    <ArrowForward color='primary' />
                  </Else>
                </If>
              </ListItemIcon>
            </Grid>
          </Grid>
        </ListItem>
      </List>
      <Divider classes={{ root: classes.divider }} />
    </>
  );
});
