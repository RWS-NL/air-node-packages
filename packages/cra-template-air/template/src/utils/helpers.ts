import { capitilizeFirstLetter, snakeToCamelCase } from '@rws-air/utils';
import { Oauth2Permissions, UserPrivileges } from 'store/oauth2/oauth2Types';

export const noOp = () => undefined;

export const transformPrivileges = (privileges: UserPrivileges) =>
  privileges
    .map((entry) => `can${capitilizeFirstLetter(snakeToCamelCase(entry.toLowerCase()))}`)
    .reduce<Oauth2Permissions>((prevVal, curVal) => ({ [curVal]: true, ...prevVal }), {});
