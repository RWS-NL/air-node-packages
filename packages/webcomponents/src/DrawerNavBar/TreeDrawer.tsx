import Box from '@material-ui/core/Box';
import Divider from '@material-ui/core/Divider';
import Drawer from '@material-ui/core/Drawer';
import { createStyles, makeStyles, Theme } from '@material-ui/core/styles';
import clsx from 'clsx';
import React, { memo } from 'react';
import { DetailsPane } from './DetailsPane';
import { TreeAndNavProps } from './DrawerProps';
import { TreeDrawerHeader } from './TreeDrawerHeader';

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    treeDrawer: {
      flexShrink: 0,
      whiteSpace: 'nowrap'
    },
    treeDrawerOpen: {
      width: ({ treeDrawer }: TreeAndNavProps) => treeDrawer.width,
      transition: theme.transitions.create('width', {
        easing: theme.transitions.easing.sharp,
        duration: theme.transitions.duration.enteringScreen
      })
    },
    treeDrawerClose: {
      transition: theme.transitions.create('width', {
        easing: theme.transitions.easing.sharp,
        duration: theme.transitions.duration.leavingScreen
      }),
      overflowX: 'hidden',
      width: theme.spacing(7) + 1,
      [theme.breakpoints.up('sm')]: {
        width: theme.spacing(9) + 1
      }
    },
    drawerPaper: {
      marginTop: theme.spacing(6),
      [theme.breakpoints.down('sm')]: {
        marginTop: theme.spacing(6)
      }
    },
    treeDrawerPaper: {
      paddingRight: ({ treeDrawer }: TreeAndNavProps) => theme.spacing(treeDrawer.open ? 1 : 0),
      left: ({ navigationDrawer }: TreeAndNavProps) =>
        navigationDrawer.open ? navigationDrawer.width + 2 : theme.spacing(7) + 2,
      [theme.breakpoints.up('sm')]: {
        left: ({ navigationDrawer }: TreeAndNavProps) =>
          navigationDrawer.open ? navigationDrawer.width + 2 : theme.spacing(9) + 2
      }
    },
    resizeDivider: {
      cursor: 'col-resize',
      width: theme.spacing(1),
      zIndex: theme.zIndex.drawer + 1,
      left: ({ treeDrawer }: TreeAndNavProps) => `calc(${treeDrawer.width}px + ${theme.spacing(0.5)}px)`
    }
  })
);

export interface TreeDrawerComponentProps extends TreeAndNavProps {
  onMouseDown: (event: React.MouseEvent<HTMLHRElement>) => void;
  onTouchStart: (event: React.TouchEvent<HTMLHRElement>) => void;
}

export const TreeDrawer = memo(
  ({ treeDrawer, navigationDrawer, onMouseDown, onTouchStart, ...props }: TreeDrawerComponentProps) => {
    const classes = useStyles({ treeDrawer, navigationDrawer });

    return (
      <>
        <DetailsPane treeDrawer={treeDrawer} navigationDrawer={navigationDrawer} />
        <Drawer
          {...props}
          variant='permanent'
          className={clsx(classes.treeDrawer, {
            [classes.treeDrawerOpen]: treeDrawer.open,
            [classes.treeDrawerClose]: !treeDrawer.open
          })}
          classes={{
            paper: clsx(classes.drawerPaper, classes.treeDrawerPaper, {
              [classes.treeDrawerOpen]: treeDrawer.open,
              [classes.treeDrawerClose]: !treeDrawer.open
            })
          }}
        >
          <TreeDrawerHeader {...treeDrawer} />
          <Divider
            data-qa='draggable-divider'
            onMouseDown={onMouseDown}
            onTouchStart={onTouchStart}
            classes={{ root: classes.resizeDivider }}
            orientation='vertical'
            draggable
            absolute
            light
          />
          <Box component='div' {...treeDrawer.ContentBoxProps}>
            {treeDrawer.content}
          </Box>
        </Drawer>
      </>
    );
  }
);
