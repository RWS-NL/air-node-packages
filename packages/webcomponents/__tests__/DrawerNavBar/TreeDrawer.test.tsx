import { TreeDrawer, TreeDrawerComponentProps } from '@DrawerNavBar/TreeDrawer';
import { TreeDrawerHeader } from '@DrawerNavBar/TreeDrawerHeader';
import { ListItem } from '@material-ui/core';
import { navigationDrawerProps, treeDrawerItems, treeDrawerProps } from '@Resource/commonDrawerNavBarProps';
import { findShallowByDataQa, toBeCalled } from '@rws-air/utils';
import { shallow } from 'enzyme';
import React from 'react';

const mockOnMouseDown = jest.fn();
const mockOnTouchStart = jest.fn();

const props: TreeDrawerComponentProps = {
  treeDrawer: treeDrawerProps,
  navigationDrawer: navigationDrawerProps,
  onMouseDown: mockOnMouseDown,
  onTouchStart: mockOnTouchStart
};

describe('TreeDrawer', () => {
  describe('Component Testing', () => {
    const treeDrawer = shallow(<TreeDrawer {...props} />);
    test('GIVEN TreeDrawer THEN contains TreeDrawerHeader', () => {
      expect(treeDrawer.find(TreeDrawerHeader).exists()).toBe(true);
    });

    test('GIVEN TreeDrawer THEN contains ListItems', () => {
      expect(treeDrawer.find(ListItem)).toHaveLength(treeDrawerItems.length);
    });

    test('GIVEN mouseDown event on draggable divider THEN triggers function', () => {
      findShallowByDataQa(treeDrawer, 'draggable-divider').simulate('mousedown');

      toBeCalled(mockOnMouseDown, 1);
    });

    test('GIVEN touchStart event on draggable divider THEN triggers function', () => {
      findShallowByDataQa(treeDrawer, 'draggable-divider').simulate('touchstart');

      toBeCalled(mockOnTouchStart, 1);
    });
  });

  describe('Snapshot Testing', () => {
    test('Required Props', () => {
      const treeDrawer = shallow(<TreeDrawer {...props} />);
      expect(treeDrawer).toMatchSnapshot();
    });
  });
});
