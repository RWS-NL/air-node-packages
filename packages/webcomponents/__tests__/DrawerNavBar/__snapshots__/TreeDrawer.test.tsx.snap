// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TreeDrawer Snapshot Testing Required Props 1`] = `
<WithStyles(ForwardRef(Drawer))
  className="makeStyles-treeDrawer-1 makeStyles-treeDrawerOpen-2 makeStyles-treeDrawerOpen-10"
  classes={
    Object {
      "paper": "makeStyles-drawerPaper-4 makeStyles-treeDrawerPaper-5 makeStyles-treeDrawerPaper-11 makeStyles-treeDrawerOpen-2 makeStyles-treeDrawerOpen-10",
    }
  }
  variant="permanent"
>
  <Memo()
    content={
      <WithStyles(ForwardRef(Paper))
        elevation={2}
      >
        <WithStyles(ForwardRef(List))>
          <WithStyles(ForwardRef(ListItem))
            button={true}
          >
            <WithStyles(ForwardRef(ListItemIcon))>
              <Memo(MoveToInboxIcon) />
            </WithStyles(ForwardRef(ListItemIcon))>
            <WithStyles(ForwardRef(ListItemText))
              primary="Inbox"
              primaryTypographyProps={
                Object {
                  "color": "primary",
                  "variant": "body2",
                }
              }
            />
          </WithStyles(ForwardRef(ListItem))>
          <WithStyles(ForwardRef(ListItem))
            button={true}
          >
            <WithStyles(ForwardRef(ListItemIcon))>
              <Memo(DraftsIcon) />
            </WithStyles(ForwardRef(ListItemIcon))>
            <WithStyles(ForwardRef(ListItemText))
              primary="Drafts"
              primaryTypographyProps={
                Object {
                  "color": "primary",
                  "variant": "body2",
                }
              }
            />
          </WithStyles(ForwardRef(ListItem))>
          <WithStyles(ForwardRef(ListItem))
            button={true}
          >
            <WithStyles(ForwardRef(ListItemIcon))>
              <Memo(MoveToInboxIcon) />
            </WithStyles(ForwardRef(ListItemIcon))>
            <WithStyles(ForwardRef(ListItemText))
              primary="Inbox"
              primaryTypographyProps={
                Object {
                  "color": "primary",
                  "variant": "body2",
                }
              }
            />
          </WithStyles(ForwardRef(ListItem))>
          <WithStyles(ForwardRef(ListItem))
            button={true}
          >
            <WithStyles(ForwardRef(ListItemIcon))>
              <Memo(DraftsIcon) />
            </WithStyles(ForwardRef(ListItemIcon))>
            <WithStyles(ForwardRef(ListItemText))
              primary="Drafts"
              primaryTypographyProps={
                Object {
                  "color": "primary",
                  "variant": "body2",
                }
              }
            />
          </WithStyles(ForwardRef(ListItem))>
          <WithStyles(ForwardRef(ListItem))
            button={true}
          >
            <WithStyles(ForwardRef(ListItemIcon))>
              <Memo(MoveToInboxIcon) />
            </WithStyles(ForwardRef(ListItemIcon))>
            <WithStyles(ForwardRef(ListItemText))
              primary="Inbox"
              primaryTypographyProps={
                Object {
                  "color": "primary",
                  "variant": "body2",
                }
              }
            />
          </WithStyles(ForwardRef(ListItem))>
          <WithStyles(ForwardRef(ListItem))
            button={true}
          >
            <WithStyles(ForwardRef(ListItemIcon))>
              <Memo(DraftsIcon) />
            </WithStyles(ForwardRef(ListItemIcon))>
            <WithStyles(ForwardRef(ListItemText))
              primary="Drafts"
              primaryTypographyProps={
                Object {
                  "color": "primary",
                  "variant": "body2",
                }
              }
            />
          </WithStyles(ForwardRef(ListItem))>
          <WithStyles(ForwardRef(ListItem))
            button={true}
          >
            <WithStyles(ForwardRef(ListItemIcon))>
              <Memo(MoveToInboxIcon) />
            </WithStyles(ForwardRef(ListItemIcon))>
            <WithStyles(ForwardRef(ListItemText))
              primary="Inbox"
              primaryTypographyProps={
                Object {
                  "color": "primary",
                  "variant": "body2",
                }
              }
            />
          </WithStyles(ForwardRef(ListItem))>
          <WithStyles(ForwardRef(ListItem))
            button={true}
          >
            <WithStyles(ForwardRef(ListItemIcon))>
              <Memo(DraftsIcon) />
            </WithStyles(ForwardRef(ListItemIcon))>
            <WithStyles(ForwardRef(ListItemText))
              primary="Drafts"
              primaryTypographyProps={
                Object {
                  "color": "primary",
                  "variant": "body2",
                }
              }
            />
          </WithStyles(ForwardRef(ListItem))>
          <WithStyles(ForwardRef(ListItem))
            button={true}
          >
            <WithStyles(ForwardRef(ListItemIcon))>
              <Memo(MoveToInboxIcon) />
            </WithStyles(ForwardRef(ListItemIcon))>
            <WithStyles(ForwardRef(ListItemText))
              primary="Inbox"
              primaryTypographyProps={
                Object {
                  "color": "primary",
                  "variant": "body2",
                }
              }
            />
          </WithStyles(ForwardRef(ListItem))>
          <WithStyles(ForwardRef(ListItem))
            button={true}
          >
            <WithStyles(ForwardRef(ListItemIcon))>
              <Memo(DraftsIcon) />
            </WithStyles(ForwardRef(ListItemIcon))>
            <WithStyles(ForwardRef(ListItemText))
              primary="Drafts"
              primaryTypographyProps={
                Object {
                  "color": "primary",
                  "variant": "body2",
                }
              }
            />
          </WithStyles(ForwardRef(ListItem))>
          <WithStyles(ForwardRef(ListItem))
            button={true}
          >
            <WithStyles(ForwardRef(ListItemIcon))>
              <Memo(MoveToInboxIcon) />
            </WithStyles(ForwardRef(ListItemIcon))>
            <WithStyles(ForwardRef(ListItemText))
              primary="Inbox"
              primaryTypographyProps={
                Object {
                  "color": "primary",
                  "variant": "body2",
                }
              }
            />
          </WithStyles(ForwardRef(ListItem))>
          <WithStyles(ForwardRef(ListItem))
            button={true}
          >
            <WithStyles(ForwardRef(ListItemIcon))>
              <Memo(DraftsIcon) />
            </WithStyles(ForwardRef(ListItemIcon))>
            <WithStyles(ForwardRef(ListItemText))
              primary="Drafts"
              primaryTypographyProps={
                Object {
                  "color": "primary",
                  "variant": "body2",
                }
              }
            />
          </WithStyles(ForwardRef(ListItem))>
          <WithStyles(ForwardRef(ListItem))
            button={true}
          >
            <WithStyles(ForwardRef(ListItemIcon))>
              <Memo(MoveToInboxIcon) />
            </WithStyles(ForwardRef(ListItemIcon))>
            <WithStyles(ForwardRef(ListItemText))
              primary="Inbox"
              primaryTypographyProps={
                Object {
                  "color": "primary",
                  "variant": "body2",
                }
              }
            />
          </WithStyles(ForwardRef(ListItem))>
          <WithStyles(ForwardRef(ListItem))
            button={true}
          >
            <WithStyles(ForwardRef(ListItemIcon))>
              <Memo(DraftsIcon) />
            </WithStyles(ForwardRef(ListItemIcon))>
            <WithStyles(ForwardRef(ListItemText))
              primary="Drafts"
              primaryTypographyProps={
                Object {
                  "color": "primary",
                  "variant": "body2",
                }
              }
            />
          </WithStyles(ForwardRef(ListItem))>
          <WithStyles(ForwardRef(ListItem))
            button={true}
          >
            <WithStyles(ForwardRef(ListItemIcon))>
              <Memo(MoveToInboxIcon) />
            </WithStyles(ForwardRef(ListItemIcon))>
            <WithStyles(ForwardRef(ListItemText))
              primary="Inbox"
              primaryTypographyProps={
                Object {
                  "color": "primary",
                  "variant": "body2",
                }
              }
            />
          </WithStyles(ForwardRef(ListItem))>
          <WithStyles(ForwardRef(ListItem))
            button={true}
          >
            <WithStyles(ForwardRef(ListItemIcon))>
              <Memo(DraftsIcon) />
            </WithStyles(ForwardRef(ListItemIcon))>
            <WithStyles(ForwardRef(ListItemText))
              primary="Drafts"
              primaryTypographyProps={
                Object {
                  "color": "primary",
                  "variant": "body2",
                }
              }
            />
          </WithStyles(ForwardRef(ListItem))>
          <WithStyles(ForwardRef(ListItem))
            button={true}
          >
            <WithStyles(ForwardRef(ListItemIcon))>
              <Memo(MoveToInboxIcon) />
            </WithStyles(ForwardRef(ListItemIcon))>
            <WithStyles(ForwardRef(ListItemText))
              primary="Inbox"
              primaryTypographyProps={
                Object {
                  "color": "primary",
                  "variant": "body2",
                }
              }
            />
          </WithStyles(ForwardRef(ListItem))>
          <WithStyles(ForwardRef(ListItem))
            button={true}
          >
            <WithStyles(ForwardRef(ListItemIcon))>
              <Memo(DraftsIcon) />
            </WithStyles(ForwardRef(ListItemIcon))>
            <WithStyles(ForwardRef(ListItemText))
              primary="Drafts"
              primaryTypographyProps={
                Object {
                  "color": "primary",
                  "variant": "body2",
                }
              }
            />
          </WithStyles(ForwardRef(ListItem))>
          <WithStyles(ForwardRef(ListItem))
            button={true}
          >
            <WithStyles(ForwardRef(ListItemIcon))>
              <Memo(MoveToInboxIcon) />
            </WithStyles(ForwardRef(ListItemIcon))>
            <WithStyles(ForwardRef(ListItemText))
              primary="Inbox"
              primaryTypographyProps={
                Object {
                  "color": "primary",
                  "variant": "body2",
                }
              }
            />
          </WithStyles(ForwardRef(ListItem))>
          <WithStyles(ForwardRef(ListItem))
            button={true}
          >
            <WithStyles(ForwardRef(ListItemIcon))>
              <Memo(DraftsIcon) />
            </WithStyles(ForwardRef(ListItemIcon))>
            <WithStyles(ForwardRef(ListItemText))
              primary="Drafts"
              primaryTypographyProps={
                Object {
                  "color": "primary",
                  "variant": "body2",
                }
              }
            />
          </WithStyles(ForwardRef(ListItem))>
          <WithStyles(ForwardRef(ListItem))
            button={true}
          >
            <WithStyles(ForwardRef(ListItemIcon))>
              <Memo(MoveToInboxIcon) />
            </WithStyles(ForwardRef(ListItemIcon))>
            <WithStyles(ForwardRef(ListItemText))
              primary="Inbox"
              primaryTypographyProps={
                Object {
                  "color": "primary",
                  "variant": "body2",
                }
              }
            />
          </WithStyles(ForwardRef(ListItem))>
        </WithStyles(ForwardRef(List))>
      </WithStyles(ForwardRef(Paper))>
    }
    currentDropdownValue="ONE"
    dropdownValues={
      Array [
        Object {
          "label": "ONE",
          "value": "ONE",
        },
        Object {
          "label": "TWO",
          "value": "TWO",
        },
        Object {
          "label": "THREE",
          "value": "THREE",
        },
        Object {
          "label": "FOUR",
          "value": "FOUR",
        },
        Object {
          "label": "FIVE",
          "value": "FIVE",
        },
        Object {
          "label": "SIX",
          "value": "SIX",
        },
      ]
    }
    hanbdleDropdownChange={[MockFunction]}
    minWidth={400}
    onBorderDrag={[MockFunction]}
    open={true}
    toggleDrawer={[MockFunction]}
    tooltipText="Hier kunt u andere type bomen selecteren. Druk op het menu links van deze knop en selecteer dan uw gewenste boom type."
    width={240}
  />
  <WithStyles(ForwardRef(Divider))
    absolute={true}
    classes={
      Object {
        "root": "makeStyles-resizeDivider-6 makeStyles-resizeDivider-12",
      }
    }
    data-qa="draggable-divider"
    draggable={true}
    light={true}
    onMouseDown={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
    onTouchStart={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
    orientation="vertical"
  />
  <Styled(MuiBox)
    component="div"
  >
    <WithStyles(ForwardRef(Paper))
      elevation={2}
    >
      <WithStyles(ForwardRef(List))>
        <WithStyles(ForwardRef(ListItem))
          button={true}
          key="0"
        >
          <WithStyles(ForwardRef(ListItemIcon))>
            <MoveToInboxIcon />
          </WithStyles(ForwardRef(ListItemIcon))>
          <WithStyles(ForwardRef(ListItemText))
            primary="Inbox"
            primaryTypographyProps={
              Object {
                "color": "primary",
                "variant": "body2",
              }
            }
          />
        </WithStyles(ForwardRef(ListItem))>
        <WithStyles(ForwardRef(ListItem))
          button={true}
          key="1"
        >
          <WithStyles(ForwardRef(ListItemIcon))>
            <DraftsIcon />
          </WithStyles(ForwardRef(ListItemIcon))>
          <WithStyles(ForwardRef(ListItemText))
            primary="Drafts"
            primaryTypographyProps={
              Object {
                "color": "primary",
                "variant": "body2",
              }
            }
          />
        </WithStyles(ForwardRef(ListItem))>
        <WithStyles(ForwardRef(ListItem))
          button={true}
          key="2"
        >
          <WithStyles(ForwardRef(ListItemIcon))>
            <MoveToInboxIcon />
          </WithStyles(ForwardRef(ListItemIcon))>
          <WithStyles(ForwardRef(ListItemText))
            primary="Inbox"
            primaryTypographyProps={
              Object {
                "color": "primary",
                "variant": "body2",
              }
            }
          />
        </WithStyles(ForwardRef(ListItem))>
        <WithStyles(ForwardRef(ListItem))
          button={true}
          key="3"
        >
          <WithStyles(ForwardRef(ListItemIcon))>
            <DraftsIcon />
          </WithStyles(ForwardRef(ListItemIcon))>
          <WithStyles(ForwardRef(ListItemText))
            primary="Drafts"
            primaryTypographyProps={
              Object {
                "color": "primary",
                "variant": "body2",
              }
            }
          />
        </WithStyles(ForwardRef(ListItem))>
        <WithStyles(ForwardRef(ListItem))
          button={true}
          key="4"
        >
          <WithStyles(ForwardRef(ListItemIcon))>
            <MoveToInboxIcon />
          </WithStyles(ForwardRef(ListItemIcon))>
          <WithStyles(ForwardRef(ListItemText))
            primary="Inbox"
            primaryTypographyProps={
              Object {
                "color": "primary",
                "variant": "body2",
              }
            }
          />
        </WithStyles(ForwardRef(ListItem))>
        <WithStyles(ForwardRef(ListItem))
          button={true}
          key="5"
        >
          <WithStyles(ForwardRef(ListItemIcon))>
            <DraftsIcon />
          </WithStyles(ForwardRef(ListItemIcon))>
          <WithStyles(ForwardRef(ListItemText))
            primary="Drafts"
            primaryTypographyProps={
              Object {
                "color": "primary",
                "variant": "body2",
              }
            }
          />
        </WithStyles(ForwardRef(ListItem))>
        <WithStyles(ForwardRef(ListItem))
          button={true}
          key="6"
        >
          <WithStyles(ForwardRef(ListItemIcon))>
            <MoveToInboxIcon />
          </WithStyles(ForwardRef(ListItemIcon))>
          <WithStyles(ForwardRef(ListItemText))
            primary="Inbox"
            primaryTypographyProps={
              Object {
                "color": "primary",
                "variant": "body2",
              }
            }
          />
        </WithStyles(ForwardRef(ListItem))>
        <WithStyles(ForwardRef(ListItem))
          button={true}
          key="7"
        >
          <WithStyles(ForwardRef(ListItemIcon))>
            <DraftsIcon />
          </WithStyles(ForwardRef(ListItemIcon))>
          <WithStyles(ForwardRef(ListItemText))
            primary="Drafts"
            primaryTypographyProps={
              Object {
                "color": "primary",
                "variant": "body2",
              }
            }
          />
        </WithStyles(ForwardRef(ListItem))>
        <WithStyles(ForwardRef(ListItem))
          button={true}
          key="8"
        >
          <WithStyles(ForwardRef(ListItemIcon))>
            <MoveToInboxIcon />
          </WithStyles(ForwardRef(ListItemIcon))>
          <WithStyles(ForwardRef(ListItemText))
            primary="Inbox"
            primaryTypographyProps={
              Object {
                "color": "primary",
                "variant": "body2",
              }
            }
          />
        </WithStyles(ForwardRef(ListItem))>
        <WithStyles(ForwardRef(ListItem))
          button={true}
          key="9"
        >
          <WithStyles(ForwardRef(ListItemIcon))>
            <DraftsIcon />
          </WithStyles(ForwardRef(ListItemIcon))>
          <WithStyles(ForwardRef(ListItemText))
            primary="Drafts"
            primaryTypographyProps={
              Object {
                "color": "primary",
                "variant": "body2",
              }
            }
          />
        </WithStyles(ForwardRef(ListItem))>
        <WithStyles(ForwardRef(ListItem))
          button={true}
          key="10"
        >
          <WithStyles(ForwardRef(ListItemIcon))>
            <MoveToInboxIcon />
          </WithStyles(ForwardRef(ListItemIcon))>
          <WithStyles(ForwardRef(ListItemText))
            primary="Inbox"
            primaryTypographyProps={
              Object {
                "color": "primary",
                "variant": "body2",
              }
            }
          />
        </WithStyles(ForwardRef(ListItem))>
        <WithStyles(ForwardRef(ListItem))
          button={true}
          key="11"
        >
          <WithStyles(ForwardRef(ListItemIcon))>
            <DraftsIcon />
          </WithStyles(ForwardRef(ListItemIcon))>
          <WithStyles(ForwardRef(ListItemText))
            primary="Drafts"
            primaryTypographyProps={
              Object {
                "color": "primary",
                "variant": "body2",
              }
            }
          />
        </WithStyles(ForwardRef(ListItem))>
        <WithStyles(ForwardRef(ListItem))
          button={true}
          key="12"
        >
          <WithStyles(ForwardRef(ListItemIcon))>
            <MoveToInboxIcon />
          </WithStyles(ForwardRef(ListItemIcon))>
          <WithStyles(ForwardRef(ListItemText))
            primary="Inbox"
            primaryTypographyProps={
              Object {
                "color": "primary",
                "variant": "body2",
              }
            }
          />
        </WithStyles(ForwardRef(ListItem))>
        <WithStyles(ForwardRef(ListItem))
          button={true}
          key="13"
        >
          <WithStyles(ForwardRef(ListItemIcon))>
            <DraftsIcon />
          </WithStyles(ForwardRef(ListItemIcon))>
          <WithStyles(ForwardRef(ListItemText))
            primary="Drafts"
            primaryTypographyProps={
              Object {
                "color": "primary",
                "variant": "body2",
              }
            }
          />
        </WithStyles(ForwardRef(ListItem))>
        <WithStyles(ForwardRef(ListItem))
          button={true}
          key="14"
        >
          <WithStyles(ForwardRef(ListItemIcon))>
            <MoveToInboxIcon />
          </WithStyles(ForwardRef(ListItemIcon))>
          <WithStyles(ForwardRef(ListItemText))
            primary="Inbox"
            primaryTypographyProps={
              Object {
                "color": "primary",
                "variant": "body2",
              }
            }
          />
        </WithStyles(ForwardRef(ListItem))>
        <WithStyles(ForwardRef(ListItem))
          button={true}
          key="15"
        >
          <WithStyles(ForwardRef(ListItemIcon))>
            <DraftsIcon />
          </WithStyles(ForwardRef(ListItemIcon))>
          <WithStyles(ForwardRef(ListItemText))
            primary="Drafts"
            primaryTypographyProps={
              Object {
                "color": "primary",
                "variant": "body2",
              }
            }
          />
        </WithStyles(ForwardRef(ListItem))>
        <WithStyles(ForwardRef(ListItem))
          button={true}
          key="16"
        >
          <WithStyles(ForwardRef(ListItemIcon))>
            <MoveToInboxIcon />
          </WithStyles(ForwardRef(ListItemIcon))>
          <WithStyles(ForwardRef(ListItemText))
            primary="Inbox"
            primaryTypographyProps={
              Object {
                "color": "primary",
                "variant": "body2",
              }
            }
          />
        </WithStyles(ForwardRef(ListItem))>
        <WithStyles(ForwardRef(ListItem))
          button={true}
          key="17"
        >
          <WithStyles(ForwardRef(ListItemIcon))>
            <DraftsIcon />
          </WithStyles(ForwardRef(ListItemIcon))>
          <WithStyles(ForwardRef(ListItemText))
            primary="Drafts"
            primaryTypographyProps={
              Object {
                "color": "primary",
                "variant": "body2",
              }
            }
          />
        </WithStyles(ForwardRef(ListItem))>
        <WithStyles(ForwardRef(ListItem))
          button={true}
          key="18"
        >
          <WithStyles(ForwardRef(ListItemIcon))>
            <MoveToInboxIcon />
          </WithStyles(ForwardRef(ListItemIcon))>
          <WithStyles(ForwardRef(ListItemText))
            primary="Inbox"
            primaryTypographyProps={
              Object {
                "color": "primary",
                "variant": "body2",
              }
            }
          />
        </WithStyles(ForwardRef(ListItem))>
        <WithStyles(ForwardRef(ListItem))
          button={true}
          key="19"
        >
          <WithStyles(ForwardRef(ListItemIcon))>
            <DraftsIcon />
          </WithStyles(ForwardRef(ListItemIcon))>
          <WithStyles(ForwardRef(ListItemText))
            primary="Drafts"
            primaryTypographyProps={
              Object {
                "color": "primary",
                "variant": "body2",
              }
            }
          />
        </WithStyles(ForwardRef(ListItem))>
        <WithStyles(ForwardRef(ListItem))
          button={true}
          key="20"
        >
          <WithStyles(ForwardRef(ListItemIcon))>
            <MoveToInboxIcon />
          </WithStyles(ForwardRef(ListItemIcon))>
          <WithStyles(ForwardRef(ListItemText))
            primary="Inbox"
            primaryTypographyProps={
              Object {
                "color": "primary",
                "variant": "body2",
              }
            }
          />
        </WithStyles(ForwardRef(ListItem))>
      </WithStyles(ForwardRef(List))>
    </WithStyles(ForwardRef(Paper))>
  </Styled(MuiBox)>
</WithStyles(ForwardRef(Drawer))>
`;
